### DISCUSSION TITLE
IS2 on NRF5340 MCK frequency

### DISCUSSION THREAD

**Discussion by farukhalilagicc** [zephyrproject-rtos/zephyr#61158] [Q&A]
Hello. I want to sample audio with 44.1 kHz sample rate, and that results in 1.4MHz BCL. This configuration results in MCK speed that is also 1.4 MHZ. My audio codec expects 11.2MHz MCK frequency. So i changed i2s_nrfx.c driver, and  got info via uart that my sample rate is 44.1kHz and MCK speed is 11.2MHz and that satisfies my specification. But when I inspected signal with osciloscope, clock is really distroted. Can anyone help me ?

static void find_suitable_clock(const struct i2s_nrfx_drv_cfg *drv_cfg,
                                nrfx_i2s_config_t *config,
                                const struct i2s_config *i2s_cfg)
{
        static const struct {
                uint16_t        ratio_val;
                nrf_i2s_ratio_t ratio_enum;
        } ratios[] = {
 //               {  32, NRF_I2S_RATIO_32X },
 //               {  48, NRF_I2S_RATIO_48X },
 //               {  64, NRF_I2S_RATIO_64X },
 //               {  96, NRF_I2S_RATIO_96X },
 //               { 128, NRF_I2S_RATIO_128X },
 //               { 192, NRF_I2S_RATIO_192X },
                  { 256, NRF_I2S_RATIO_256X },
//               { 384, NRF_I2S_RATIO_384X },
//                { 512, NRF_I2S_RATIO_512X }
        };

**Comment by koffes** [comment]
Do you have a capture of your oscilloscope we can view? Are you using a standard nRF5340DK or some other hardware? Note that 11.2 MHz is quite high, so if your scope probes have high capacitance that will round the signals.
